### PR TITLE
new HUA id on dev, add category to race and ethnicity output

### PR DESCRIPTION
**Pull request by longshuicy**
@ncsa-mo  i created a draft PR for you. this way it's easier for me to see the change.

### CODE REVIEW

File: pyincore/utils/popdisloutputprocess.py [hunk 1]
```diff
@@ -130,33 +132,39 @@ def pd_by_race(self, filename_json=None):
...
                 hua_disl.append(0)
-        hua_disl.append(int(sum(hua_disl[1:])))
+        hua_disl.append(int(sum(hua_disl)))
 
         pd_disl = []
-        for i in range(len(race_categories)):
-            pd_disl.append(int(huapd["numprec"].where(huapd["hud_re"] == str(i)).sum()))
-        pd_disl.append(int(sum(pd_disl[1:])))
+        for i in range(len(race_categories) - 1):
+            # vacant HU do not dislocate
```
Comment (longshuicy) [inline pyincore/utils/popdisloutputprocess.py:147]: this is not true.... i think they still dislocate. Otherwise the number doesn't add up

Comment (ncsa-mo) [reply pyincore/utils/popdisloutputprocess.py]: I changed that but it does not make sense.

**Comment by longshuicy** [comment]
Could you also change the json key that has "%_of" to "percentage_of" or "percent_of"?
This is related to this PR: https://github.com/IN-CORE/incore-playbook/pull/20

File: pyincore/utils/popdisloutputprocess.py [hunk 2]
```diff
@@ -17,35 +17,61 @@ class PopDislOutputProcess:
         pop_disl_result (obj): IN-CORE dataset for Joplin Population Dislocation (PD) results.
         pop_disl_result_path (obj): A fallback for the case that Joplin PD object is not provided.
             For example a user wants to directly pass in csv files, a path to PD results.
-        filter_on (bool): A flag to filter all data, default True counts only Joplin buildings
+        filter_name (str): A string to filter data by name, default empty. Example: filter_name="Joplin" for Joplin
+            inventory, other is Duquesne etc. Name must be valid.
+        filter_on (bool): A flag to filter all data, default True counts only Joplin buildings.
```
Comment (navarroc) [inline pyincore/utils/popdisloutputprocess.py:22]: Looks like this got renamed, can you fix this?

File: pyincore/utils/popdisloutputprocess.py [hunk 3]
```diff
@@ -17,35 +17,61 @@ class PopDislOutputProcess:
...
-                       "%_household_dislocated",
+                       "percent_household_dislocated",
                        "population_dislocated",
                        "total_population",
-                       "%_population_dislocated"
+                       "percent_population_dislocated"
                        ]
 
-    def __init__(self, pop_disl_result, pop_disl_result_path=None, filter_on=True):
+    def __init__(self, pop_disl_result, pop_disl_result_path=None, filter_name=None, filter_guid=True, vacant_disl=True):
```
Comment (navarroc) [inline pyincore/utils/popdisloutputprocess.py:35]: filter_guid is not in the method Args list.

**Comment by navarroc** [comment]
How can I test the two different methods? I tried running the test_popdisloutputprocess.py setting vacant_disl to True and False, but I don't see any differences. Am I testing it right?

**Comment by navarroc** [comment]
> How can I test the two different methods? I tried running the test_popdisloutputprocess.py setting vacant_disl to True and False, but I don't see any differences. Am I testing it right?

Ignore this, I uncommented the wrong block of code :(

**Comment by navarroc** [comment]
This change looks good except for the documentation changes that need to be done. I can approve once that's done.

**Review by navarroc — APPROVED** [review]

**Review by ncsa-mo — APPROVED** [review]

**Pull request merged**